### ISSUE TITLE
SslKafkaConsumerTest and SaslKafkaConsumerTest are both flaky

### ISSUE THREAD

**Issue by famod** [quarkusio/quarkus#16371]
Both fail frequently these days with `java.net.SocketTimeoutException: Read timed out`, e.g: https://github.com/quarkusio/quarkus/pull/16360#issuecomment-815985649

Before the failure, there many warnings like:
```
2021-04-08T16:34:51.7430524Z 2021-04-08 16:34:51,308 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | test-ssl-producer) [Producer clientId=test-ssl-producer] Error while fetching metadata with correlation id 867 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7445567Z 2021-04-08 16:34:51,308 WARN  [org.apa.kaf.cli.NetworkClient] (executor-thread-1) [Consumer clientId=consumer-test-consumer-1, groupId=test-consumer] Error while fetching metadata with correlation id 283 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7461688Z 2021-04-08 16:34:51,411 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | test-ssl-producer) [Producer clientId=test-ssl-producer] Error while fetching metadata with correlation id 868 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7476286Z 2021-04-08 16:34:51,412 WARN  [org.apa.kaf.cli.NetworkClient] (executor-thread-1) [Consumer clientId=consumer-test-consumer-1, groupId=test-consumer] Error while fetching metadata with correlation id 284 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
```
Further up in the log:
```
2021-04-08T16:34:51.5863238Z 2021-04-08 16:34:46,520 WARN  [kaf.con.RequestSendThread] (Controller-1-to-broker-1-send-thread) [RequestSendThread controllerId=1] Controller 1's connection to broker fv-az132-838.internal.cloudapp.net:19099 (id: 1 rack: null) was unsuccessful: java.net.SocketTimeoutException: Failed to connect within 30000 ms
2021-04-08T16:34:51.5866133Z 	at kafka.controller.RequestSendThread.brokerReady(ControllerChannelManager.scala:293)
2021-04-08T16:34:51.5871267Z 	at kafka.controller.RequestSendThread.doWork(ControllerChannelManager.scala:246)
2021-04-08T16:34:51.5875378Z 	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:96)
```
And this at beginning of the test doesn't look healthy either:
```
2021-04-08T16:33:19.7337164Z 2021-04-08 16:33:13,291 WARN  [kaf.ser.BrokerMetadataCheckpoint] (pool-5-thread-1) No meta.properties file under dir /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-ssl/kafka/broker1/meta.properties
2021-04-08T16:33:19.7340785Z 2021-04-08 16:33:16,231 INFO  [io.deb.kaf.KafkaServer] (pool-5-thread-1) Started Kafka server 1 at localhost:19099 with storage in /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-ssl/kafka/broker1
2021-04-08T16:33:19.7344253Z 2021-04-08 16:33:16,232 INFO  [sta.cha.logger] (controller-event-thread) [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers Set(1) for 0 partitions
2021-04-08T16:33:19.7346875Z 2021-04-08 16:33:16,397 WARN  [kaf.ser.BrokerMetadataCheckpoint] (pool-5-thread-1) No meta.properties file under dir /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-sasl/kafka/broker1/meta.properties
2021-04-08T16:33:19.7349939Z 2021-04-08 16:33:16,845 INFO  [sta.cha.logger] (controller-event-thread) [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers Set(1) for 0 partitions
2021-04-08T16:33:19.7354184Z 2021-04-08 16:33:16,854 WARN  [org.apa.kaf.com.uti.AppInfoParser] (pool-5-thread-1) Error registering AppInfo mbean: javax.management.InstanceAlreadyExistsException: kafka.server:type=app-info,id=1
2021-04-08T16:33:19.7358057Z 	at java.management/com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:436)
2021-04-08T16:33:19.7363800Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1855)
2021-04-08T16:33:19.7367574Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:955)
2021-04-08T16:33:19.7371146Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:890)
2021-04-08T16:33:19.7374303Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:320)
2021-04-08T16:33:19.7378691Z 	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
2021-04-08T16:33:19.7381955Z 	at org.apache.kafka.common.utils.AppInfoParser.registerAppInfo(AppInfoParser.java:64)
2021-04-08T16:33:19.7383571Z 	at kafka.server.KafkaServer.startup(KafkaServer.scala:393)
2021-04-08T16:33:19.7385533Z 	at io.debezium.kafka.KafkaServer.startup(KafkaServer.java:220)
2021-04-08T16:33:19.7387673Z 	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4772)
2021-04-08T16:33:19.7389727Z 	at io.debezium.kafka.KafkaCluster.startup(KafkaCluster.java:250)
2021-04-08T16:33:19.7391603Z 	at io.quarkus.it.kafka.KafkaSASLTestResource.start(KafkaSASLTestResource.java:49)
2021-04-08T16:33:19.7399148Z 	at io.quarkus.test.common.TestResourceManager.lambda$start$1(TestResourceManager.java:107)
2021-04-08T16:33:19.7427944Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2021-04-08T16:33:19.7442008Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-04-08T16:33:19.7443548Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-04-08T16:33:19.7446072Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-04-08T16:33:19.7447292Z 	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by famod** [comment]
So it seems these tests can even cause a build to hang until it is aborted due to timeout: https://github.com/quarkusio/quarkus/pull/16249/checks?check_run_id=2310380201

**Comment by cescoffier** [comment]
Should be fixed by #16435

**Comment by famod** [comment]
Thanks @cescoffier! I'll close this one. In case the problem pops up again we can always reopen this issue.
